src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Presentation.Api.OData/CountriesController.g.cs

```diff
@@ -360,6 +360,52 @@ public async Task<ActionResult> DeleteRefToWorkplaces([FromRoute] System.Int64 k
         return NoContent();
     }
     
+    [HttpDelete("api/Countries/{key}/Workplaces/{relatedKey}")]
+    public async Task<ActionResult> DeleteToWorkplaces([FromRoute] System.Int64 key, [FromRoute] System.UInt32 relatedKey)
+    {
+        if (!ModelState.IsValid)
+        {
+            return BadRequest(ModelState);
+        }
+        
+        var related = (await _mediator.Send(new GetCountryByIdQuery(key))).SelectMany(x => x.Workplaces).Any(x => x.Id == relatedKey);
+        if (!related)
+        {
+            return NotFound();
+        }
+        
+        var etag = Request.GetDecodedEtagHeader();
+        var deleted = await _mediator.Send(new DeleteWorkplaceByIdCommand(relatedKey, etag));
+        if (!deleted)
+        {
+            return NotFound();
+        }
+        
+        return NoContent();
+    }
+    
+    [HttpDelete("api/Countries/{key}/Workplaces")]
+    public async Task<ActionResult> DeleteToWorkplaces([FromRoute] System.Int64 key)
+    {
+        if (!ModelState.IsValid)
+        {
+            return BadRequest(ModelState);
+        }
+        
+        var related = (await _mediator.Send(new GetCountryByIdQuery(key))).Select(x => x.Workplaces).SingleOrDefault();
+        if (related == null)
+        {
+            return NotFound();
+        }
+        
+        var etag = Request.GetDecodedEtagHeader();
+        foreach(var item in related)
+        {
+            await _mediator.Send(new DeleteWorkplaceByIdCommand(item.Id, etag));
```

Comment:
@rochar should we create a transaction or a new command to handle this case? What if any related entity from the list is not removed for any reasons?

Reply:
yes, any "workflow", should translate to a single command, and therefore encapsulated in a single transaction

we may reuse DeleteWorkplaceByIdCommand and make it DeleteWorkplacesByIdCommand, so it can receive multiple or just one, what do y thing?

I will merge we can do this in the next PR